superset/viz.py

```diff
@@ -57,6 +57,7 @@ def __init__(self, datasource, form_data):
             'token', 'token_' + uuid.uuid4().hex[:8])
         self.metrics = self.form_data.get('metrics') or []
         self.groupby = self.form_data.get('groupby') or []
+        self.annotation_layers = []
```

Comment:
I don't think annotations should be part the viz object. For most viz types that I can think of annotations don't really make sense. If you make annotations accessible from a json endpoint they can be loaded client side. This would also make the responsiveness be better, annotations could be loaded async independently of the data.

Reply:
Here we decided to package the annotations from the backend along with the data payload. The visualizations' render method can then assume that all of the data + annotations + context is present when rendering. Note that the annotation query is fully indexed and should take milliseconds assuming a small-ish number of annotations for the time range.

I feel like if we wanted to go async on the annotations, we'd need to change the visualization's API to define 2 different methods and somehow guarantee to call the annotation one second. It makes things much more complicated on that side of things.

Reply:
https://github.com/apache/incubator-superset/pull/3518/commits/34ec8b35a8eca6a6cef070387aa1c19bd4a972a1#diff-a85def8afdc240ac598637a26a7cdbbc I probably clean this up a bit more, but this makes all http requests in parallel.